system_tests/program_gas_test.go

```diff
@@ -23,6 +24,54 @@ import (
 	"github.com/offchainlabs/nitro/util/testhelpers"
 )
 
+func TestGasUsageOfHostiosThatDontHaveGoodEVMEquivalents(t *testing.T) {
+	builder := setupGasCostTest(t)
+	auth := builder.L2Info.GetDefaultTransactOpts("Owner", builder.ctx)
+	stylusProgram := deployWasm(t, builder.ctx, auth, builder.L2.Client, rustFile("hostio-test"))
+	matchSnake := regexp.MustCompile("_[a-z]")
+
+	for _, tc := range []struct {
+		hostio      string
+		expectedInk uint64
+	}{
+		{hostio: "read_args", expectedInk: 8400 + 5040},
+		{hostio: "write_result", expectedInk: 8400 + (16381+55*(10000-32))*2},
+		{hostio: "storage_cache_bytes32", expectedInk: 8400 + (13440-8400)*2},
+		{hostio: "msg_reentrant", expectedInk: 8400},
+		{hostio: "pay_for_memory_grow", expectedInk: 9320660000},
```

Comment:
let's add arguments and test with more then one value.
for read_args and write_result: different length
for pay_for_memory_grow: test both 0 and some non-0 value (100 is o.k.)

Reply:
Done.